library/swift/src/filters/Filter.swift

```diff
@@ -105,5 +105,55 @@ extension EnvoyHTTPFilter {
         }
       }
     }
+
+    if let asyncRequestFilter = filter as? AsyncRequestFilter {
+      self.setRequestFilterCallbacks = { envoyCallbacks in
+        asyncRequestFilter.setRequestFilterCallbacks(
+          RequestFilterCallbacksImpl(callbacks: envoyCallbacks)
+        )
+      }
+
+      self.onResumeRequest = { envoyHeaders, data, envoyTrailers, endStream in
+        let result = asyncRequestFilter.onResumeRequest(
+          headers: envoyHeaders.flatMap(RequestHeaders.init),
```

Comment:
```suggestion
          headers: envoyHeaders.map(RequestHeaders.init),
```

Reply:
Oh is `map` preferred/idiomatic for optional over `flatMap` now? (and did that change or did I misremember?)

Reply:
`flatMap` will unwrap the optional _and the optional returned as the result of the closure_. Since the closure returns a non-optional, there's no reason to use `flatMap` over `map`

Reply:
Thanks, that makes sense.